src/server/middleware/dev.middleware.ts

```diff
@@ -0,0 +1,71 @@
+import { resolve } from 'path';
+import webpack from 'webpack';
+import webpackDevServer from 'webpack-dev-middleware';
+import webpackHotServer from 'webpack-hot-middleware';
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+
+const publicPath = '/';
+
+/**
+ * Configures react development for the client.
+ */
+
+const config = {
+  name: 'client',
+  mode: 'development',
+  entry: [
+    'webpack-hot-middleware/client',
+    'react-hot-loader/patch',
+    './src/client/index.js',
+  ],
+  output: {
+    path: resolve(__dirname, 'build/static'),
+    filename: 'app.js',
+    publicPath,
+  },
+  target: 'web',
+  module: {
+    rules: [
+      {
+        test: /\.[jt]sx?$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'babel-loader',
+            options: {
+              plugins: ['react-hot-loader/babel'],
+            },
+          },
+          'ts-loader',
+        ],
+      },
+    ],
+  },
+  resolve: {
+    alias: {
+      'react-dom': '@hot-loader/react-dom',
+    },
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
+      title: 'Docker App',
```

Comment:
What about having this import the `name` from `package.json`?

Reply:
hmmm, I'd prefer not to use the `name` field as that's really the name for the whole project, more than the client app. But I did add an environment variable for this in 6ee2e7b (that way we can also change the displayed name in QA)

Reply:
That's a good idea!